fea-rs/src/compile/features.rs

```diff
@@ -353,6 +358,52 @@ impl ActiveFeature {
     }
 }
 
+/// Given a slice of lookupids, split them into (GPOS, GSUB)
+///
+/// In general, a feature only has either GSUB or GPOS lookups, but this is not
+/// a requirement, and in the wild we will encounter features that contain mixed
+/// lookups.
+fn split_lookups(lookups: &[LookupId]) -> (Vec<u16>, Vec<u16>) {
+    if lookups.is_empty() {
+        return (Vec::new(), Vec::new());
+    }
+
+    // in the majority of cases, a given feature only has lookups of one kind,
+    // so that is the fast path.
+    let is_gpos = matches!(lookups.first(), Some(LookupId::Gpos(_)));
+    if lookups
+        .iter()
+        .all(|x| matches!(x, LookupId::Gpos(_)) == is_gpos)
+    {
+        if is_gpos {
+            return (
+                lookups.iter().map(|x| x.to_gpos_id_or_die()).collect(),
+                Vec::new(),
+            );
+        } else {
+            return (
+                Vec::new(),
+                lookups.iter().map(|x| x.to_gsub_id_or_die()).collect(),
+            );
+        }
+    }
+
+    // the uncommon case, where we have mixed lookups;
```

Comment:
I might be missing something but the fast path doesn’t appear to actually be faster because it does two iterations over the lookups. I think we can just simplify by dropping all the code above (including the test for an empty lookup slice)?

Reply:
hmm yea, I think originally I was reusing the input array or something? happy to clean this up.